pkg/core/account/accounts_test.go

```diff
@@ -124,14 +119,78 @@ func TestAccounts(t *testing.T) {
 	*/
 }
 
+func TestAccounts_SeatedAccounts(t *testing.T) {
+	committeeSize := 20
+
+	prevCommitteeIssuers, _ := generateAccounts(committeeSize)
+	newCommitteeIssuers, _ := generateAccounts(committeeSize/2 - 1)
+
+	newCommitteeAccounts := account.NewAccounts()
+	prevCommitteeAccounts := account.NewAccounts()
+
+	idx := 0
+	for id, stake := range prevCommitteeIssuers {
+		// half of the accounts from previous committee should also be added to the new committee
+		if idx%2 == 0 || idx == 1 {
+			require.NoError(t, newCommitteeAccounts.Set(id, &account.Pool{
+				PoolStake:      stake,
+				ValidatorStake: iotago.BaseToken(stake) * 2,
+				FixedCost:      iotago.Mana(stake) * 3,
+			}))
+		}
+
+		require.NoError(t, prevCommitteeAccounts.Set(id, &account.Pool{
+			PoolStake:      stake,
+			ValidatorStake: iotago.BaseToken(stake) * 2,
+			FixedCost:      iotago.Mana(stake) * 3,
+		}))
+
+		idx++
+	}
+
+	// Add remaining accounts to new commitee accounts.
+	for id, stake := range newCommitteeIssuers {
+		// half of the accounts from previous committee should also be added to the new committee
+		require.NoError(t, newCommitteeAccounts.Set(id, &account.Pool{
+			PoolStake:      stake,
+			ValidatorStake: iotago.BaseToken(stake) * 2,
+			FixedCost:      iotago.Mana(stake) * 3,
+		}))
+	}
+
+	// Get prev committee and assign seats by default.
+	prevCommittee := prevCommitteeAccounts.SeatedAccounts()
+
+	// New committee seats should be assigned taking previous committee into account.
+	newCommittee := newCommitteeAccounts.SeatedAccounts(prevCommittee)
+
+	prevCommitteeAccounts.ForEach(func(id iotago.AccountID, _ *account.Pool) bool {
+		if seatIndex, exists := newCommittee.GetSeat(id); exists {
+			require.Equal(t, lo.Return1(prevCommittee.GetSeat(id)), seatIndex)
+		}
+
+		return true
+	})
+
+	// Make sure that size of the committee and SeatIndex values are sane.
+	require.Equal(t, newCommittee.SeatCount(), committeeSize)
+
+	newCommitteeAccounts.ForEach(func(id iotago.AccountID, _ *account.Pool) bool {
+		require.Less(t, int(lo.Return1(newCommittee.GetSeat(id))), committeeSize)
+		require.GreaterOrEqual(t, int(lo.Return1(newCommittee.GetSeat(id))), 0)
```

Comment:
What exactly are you checking here? Not easy to read

Reply:
Here I check that SeatIndices are within <0; committeeSize) range. Added a comment to explain.